pkg/reboot/rebooter.go

```diff
@@ -28,26 +32,55 @@ func NewWatchdogRebooter(wd watchdog.Watchdog, log logr.Logger) Rebooter {
 }
 
 func (r *WatchdogRebooter) Reboot() error {
-	if r.wd == nil || !r.wd.IsStarted() {
+	if r.wd == nil {
 		r.log.Info("no watchdog is present on this host, trying software reboot")
 		//we couldn't init a watchdog so far but requested to be rebooted. we issue a software reboot
-		if err := r.softwareReboot(); err != nil {
-			r.log.Error(err, "failed to run reboot command")
-			// TODO retry because of this?
-			//return err
-			return nil
-		}
+		return r.softwareReboot()
+	}
+
+	if r.isWatchdogRebootStuck() {
```

Comment:
can we move this into the `Triggered` case below?

Reply:
Yep, I agree :+1: